### PR TITLE
[Flang][OpenMP] Separate creation of work-sharing and SIMD loops, NFC

### PR DESCRIPTION
**Pull request by kparzysz**
These two constructs were both handled in `genOMP` for loop constructs. There is some shared code between the two, but there are also enough differences to separate these two cases into individual functions.

The shared code may be placed into a helper function later if needed.

Recursive lowering [1/5]

### CODE REVIEW

**Review by kiranchandramohan — APPROVED** [review]
Looks OK to me. Please wait for @skatrak.

File: flang/lib/Lower/OpenMP.cpp
Comment (skatrak) [inline flang/lib/Lower/OpenMP.cpp:3006]: I think this is worth either keeping in `genOMP` or outlining into its own function and reusing it already as part of this patch, since it's a bit too much replicated logic otherwise.

**Review by skatrak — COMMENTED** [review]
Thank you Krzysztof for this refactoring, I agree this is a good improvement to make. I just have a small comment, but otherwise it LGTM.

File: flang/lib/Lower/OpenMP.cpp
Comment (kparzysz) [reply flang/lib/Lower/OpenMP.cpp]: I have moved the conversion loop out into its own function.  The rest of the code uses a lot of variables that would have to be passed in and returned, plus the clause processing only shares some steps, but not all of them.

I hope this is ok with you.

Comment (skatrak) [reply flang/lib/Lower/OpenMP.cpp]: Yes, I think that makes sense. Thanks!

**Review by skatrak — APPROVED** [review]
LGTM! I'd just suggest waiting for the buildbot to complete before merging if you haven't already checked locally that the reordering of the `convertLoopBounds` call to after clause processing in `createSimdLoop` breaks any check-flang lit unit test. Thanks again!

**Pull request merged**
